### ISSUE TITLE
Add required asterisk indicator to form elements

### ISSUE THREAD

**Issue by jerebtw** [skeletonlabs/skeleton#765]
### Describe what feature you'd like. Pseudo-code, mockups, or screenshots of similar solutions are encouraged!

I noticed that the Mantine UI library includes a feature that adds an asterisk to form elements if the required attribute is set (see: https://mantine.dev/core/text-input/). I think this is a useful and helpful feature, and I would like to request that it be added to the Skeleton as well.

An asterisk is a common way to indicate that a form element is required, and having this feature built into the library would make it easy for developers to use it without having to add additional styling or markup themselves.

### What type of pull request would this be?

Enhancement

**Comment by endigo9740** [comment]
Hey @jerebtw, unfortunately I haven't had a chance to put my thoughts on this down in the docs or a blog post yet, but Skeleton takes a pretty hard stance against embedding inputs inside components, whenever possible. This means we don't really touch inputs aside some CSS styles and a small handful of non-standard form components (ex: slide toggle).

I'm aware we break with the norm here, but I think it's a flawed approach do this in component systems. Inputs are complex and have a LOT of attributes. See this for example:
https://www.dofactory.com/html/input-attributes

And that's just the `<input>` tag, it doesn't take into account `<select>`, etc. This means that when embedding inputs in a component you have to provide tunnels into and out of the component so folks can access the input. Svelte provides some tools for this, such as props, bind, events, and `$$restProps`. But what happens when a non-standard attribute is needed? What if a user need a specific event forward that we don't include? Then that's out of your hands as a developer - you cannot interface with the input as you need to. That's a big problem.

The TL;DR is that Skeleton doesn't really provide logic for your forms and inputs. However, I am confident that either Svelte/SvelteKit itself or a third party library exist that provide what you're looking for. I'd encourage you to look around and share if you find a good solution that we can pass on. 

But in the short term I don't see this as something that will change on our end. I hope that makes sense!
